### PR TITLE
ci: add python 3.10, gcc-10 option

### PR DESCRIPTION
**Pull request by georgebisbas**
Tested python 3.10 via docker seems ok, just spinning a python 3.10 CI

### CODE REVIEW

**Comment by mloubout** [comment]
scipy doesn't support 3.10 and numpy only in some case so until these move towards it I wouldn't even look at it.

**Comment by georgebisbas** [comment]
makes sense

**Comment by mloubout** [comment]
Looks liek scipy got a new release 3 days ago that support 3.10

File: .github/workflows/pytest-core-nompi.yml [hunk 1]
```diff
@@ -86,12 +86,12 @@ jobs:
           language: "C"
           sympy: 1.8
 
-        - name: pytest-docker-py36-gcc-omp
-          python-version: 3.6
-          os: ubuntu-18.04
+        - name: pytest-docker-py310-gcc-omp
```
Comment (mloubout) [inline .github/workflows/pytest-core-nompi.yml:89]: I wouldn't change the docker build we wanna keep a safe standard version for that one.

File: .github/workflows/pytest-core-nompi.yml [hunk 2]
```diff
@@ -29,65 +29,72 @@ jobs:
...
            pytest-ubuntu-py39-gcc9-omp,
            pytest-osx-py37-clang-omp,
-           pytest-docker-py36-gcc-omp
+           pytest-docker-py37-gcc-omp
         ]
         set: [base, adjoint]
         include:
         - name: pytest-ubuntu-py37-gcc5-omp
-          python-version: 3.7
+          python-version: '3.7'
```
Comment (FabioLuporini) [inline .github/workflows/pytest-core-nompi.yml:41]: why do we need the '...' now?

File: .github/workflows/pytest-core-nompi.yml [hunk 3]
```diff
@@ -29,65 +29,72 @@ jobs:
...
 
         - name: pytest-ubuntu-py39-gcc9-omp
-          python-version: 3.9
+          python-version: '3.9'
           os: ubuntu-20.04
           arch: "gcc-9"
           language: "openmp"
           sympy: 1.8
 
+        - name: pytest-ubuntu-py310-gcc10-omp
```
Comment (FabioLuporini) [inline .github/workflows/pytest-core-nompi.yml:82]: adding a new one, but not dropping an old one? is this PR replacing a job, or adding a new one?

**Comment by FabioLuporini** [comment]
why is the docker build also changing?

File: .github/workflows/pytest-core-nompi.yml (hunk 3 above)
Comment (georgebisbas) [reply .github/workflows/pytest-core-nompi.yml:82]: We discused in the meeting yesterday to maybe replace a 3.7 job, ` ' ' ` are needed from now for github actions.

Comment (georgebisbas) [reply .github/workflows/pytest-core-nompi.yml:82]: https://github.com/actions/setup-python/issues/160

**Comment by georgebisbas** [comment]
> why is the docker build also changing?

It had wrong name before... now it is fixed. It is using 3.7

**Review by FabioLuporini — APPROVED** [review]
LTGM

**Pull request merged**